pkgs/development/python-modules/octave-kernel/default.nix

```diff
@@ -0,0 +1,53 @@
+{
+  lib,
+  buildPythonPackage,
+  pythonOlder,
+  hatchling,
+  metakernel,
+  jupyter-client,
+  ipykernel,
+  nbconvert,
+  pytest,
+  fetchFromGitHub,
+}:
+
+buildPythonPackage rec {
+  pname = "octave_kernel";
+  version = "0.36.0";
+
+  build-system = [ hatchling ];
+  dependencies = [
+    metakernel
+    jupyter-client
+    ipykernel
+  ];
+  disables = pythonOlder "3.7";
+
+  optional-dependencies = {
+    test = [
+      pytest
+      nbconvert
+    ];
+  };
+
+  passthru = {
+    dependency-groups = { };
+  };
+
+  pyproject = true;
```

Comment:
Bit of a nitpick, but usually the first attributes in a Python module expression are ordered like so

```nix
{
  pname = "...";
  version = "...";
  pyproject = true;

  disabled = pythonOlder "3.7";

  src = { };

  build-system = [ ];

  dependencies = [ ];

  meta = { };
}
```
This isn't a hardset rule or anything though, so feel free to ignore ;)